internal/scrapertest/util.go

```diff
@@ -54,3 +54,27 @@ func sortInstrumentationLibrary(a, b pmetric.ScopeMetrics) bool {
 	}
 	return false
 }
+
+func sortResourceMetrics(a, b pmetric.ResourceMetrics) bool {
+	if a.SchemaUrl() < b.SchemaUrl() {
+		return true
+	}
+	if a.ScopeMetrics().Len() < b.ScopeMetrics().Len() {
+		return true
+	}
+	for i := 0; i < a.ScopeMetrics().Len(); i++ {
+		aSm := a.ScopeMetrics().At(i)
+		bSm := b.ScopeMetrics().At(i)
```

Comment:
What happens if `a.ScopeMetrics().Len()` > `b.ScopeMetrics().Len()`?  What is the outcome of `b.ScopeMetrics().At(i)` when `i` is out of scope?

Reply:
both the `expectedMetrics` and `actualMetrics`  will call the sort function, so they will sorted in the same rule. 

As for the `for loop` , this sort function is called after a lenght compare between `expectedMetrics` and `actualMetrics`.

this sort function also need improve, that said, current sort logic will check the lenght of  ScopeMetrics between two ResourceMetrics, if the the two ScopeMetrics have the same lenght, then it will run a loop to check the sub metric of ScopeMetrics(see the line68), however, the two ScopeMetrics——`aSm` and `bSm` are not guarantee to in a same sequence, wo can't simplily compare its lenght.

Reply:
Yes thats the scenario I was thinking of.